src/v/cloud_storage/remote_partition.cc

```diff
@@ -199,7 +200,8 @@ remote_partition::borrow_result_t remote_partition::borrow_next_segment_reader(
         }
     }
     if (iter == _segments.end()) {
-        auto path = manifest.generate_segment_path(*mit);
+        auto path = manifest.generate_segment_path(
```

Comment:
In this case the `manifest` is a spillover manifest, you're using `_manifest_view->path_provider()` here but in order to generate the spillover manifest the code is picking up the right path provider automatically. It's not incorrect but it's confusing to route all these `path_providers` to some calls and not other calls. It will be difficult for someone new to understand the code.

Reply:
We discussed offline, I'll pass the path provider directly to the `remote_partition` rather than relying on the view's path provider. Though FWIW, I think it makes sense for the remote partition to depend on async_manifest_view's -- the hierarchy at least makes sense because the remote_partition uses the view for both getting segments and cursoring over spillover manifests (done by the view internally with the view's path provider)